tests/testthat/test-brace_linter.R

```diff
@@ -2,10 +2,10 @@ test_that("brace_linter lints braces correctly", {
   open_curly_msg <- rex::rex(
     "Opening curly braces should never go on their own line"
   )
-  closed_curly_msg <- rex::rex(paste(
```

Comment:
Add `rex()` to the defaults for the linter taking care of `stop(paste())`?

Reply:
Hmm that one's called [`condition_message_linter()`](https://lintr.r-lib.org/reference/condition_message_linter.html) -- maybe it could be deprecated in favor of an `unnecessary_paste_linter()`?

Otherwise, fold it into `paste_linter()` along with `paste("...")`? https://github.com/r-lib/lintr/issues/2183#issuecomment-1723635524

Reply:
`paste_linter` sounds like the place to pur unnecessary paste calls. Maybe add an argument `allow_unnecessary = FALSE` to partially disable it.

Deprecating `condition_message_linter()` sounds like a good way to move the logic.

Reply:
#2203